sources/constraint-declaration-validation.asciidoc

```diff
@@ -1026,6 +1100,289 @@ public class SimpleOrderService extends OrderService {
+
+When passing a non-null node name to any of the receiver methods and the extractor is called in the course of container element validation (and not during cascaded validation),
+a node of kind `CONTAINER_ELEMENT` will be appended to the property path, with `Node#getName()` returning the specified name.
+If null is passed as node name, no node will be appended to the property path.
+The resulting property path will then be the same as if the constraint had been given on the container instead of a container element.
+That is desirable for pure wrapper types such as `Optional`.
+
+The container value passed to a value extractor is retrieved from the element that hosts the type argument carrying the constraint or `@Valid` annotation:
+
+[source,java]
+----
+private Map<String, @Valid @RetailOrder Order> ordersByName;
+
+public Map<@NotNull String, Order> getOrdersByName() {
+    return ordersByName;
+}
+----
+
+When validating the `@NotNull` constraint, the map as returned by the getter will be passed to the map key extractor in order to obtain the map keys.
+When validating the `@RetailOrder` constraint and performing cascaded validation,
+the map as obtained directly from the field will be passed to the map value extractor in order to obtain the map values.
+
+[[constraintdeclarationvalidationprocess-valueextractordefinitions-extractedvalue]]
+==== @ExtractedValue
+
+The `@ExtractedValue` annotation is used to denote the element extracted by a given value extractor:
+
+[source, JAVA, indent=0]
+----
+include::{validation-api-source-dir}javax/validation/valueextraction/ExtractedValue.java[lines=7..8;15..-1]
+----
+
+The `@ExtractedValue` annotation must be specified exactly once for a value extractor type.
+
+`@ExtractedValue` is to be specified on a type argument of the container type as used in the extractor definition.
+Only wildcard type arguments may be annotated with `@ExtractedValue`.
+
+[NOTE]
+====
+This implies that there may not be more than one extractor for a given generic type.
+I.e. there can be an extractor for `List<?>`, but not one for `List<String>` and one for `List<Integer>`.
+====
+
+In case an illegal value extractor definition is detected, a `ValueExtractorDefinitionException` is raised.
+
+[[constraintdeclarationvalidationprocess-valueextractordefinitions-unwrapbydefault]]
+==== @UnwrapByDefault
+
+Value extractor definitions can be marked with the `@UnwrapByDefault` annotation.
+This causes constraints to be automatically applied to the wrapped value if a constraint is found for an element handled by that extractor (see <<constraintdeclarationvalidationprocess-containerelementconstraints-implicitunwrapping>>):
+
+.`@UnwrapByDefault` annotation
+====
+[source, JAVA, indent=0]
+----
+include::{validation-api-source-dir}javax/validation/valueextraction/UnwrapByDefault.java[lines=7..8;16..-1]
```

Comment:
s/Constraint/SomeConstraint/ otherwise we think it's the actual`Constraint` type